boa_engine/src/bytecompiler/mod.rs

```diff
@@ -473,10 +549,10 @@ impl<'b, 'host> ByteCompiler<'b, 'host> {
         let index = index as usize;
 
         let bytes = target.to_ne_bytes();
-        self.code_block.bytecode[index + 1] = bytes[0];
-        self.code_block.bytecode[index + 2] = bytes[1];
-        self.code_block.bytecode[index + 3] = bytes[2];
-        self.code_block.bytecode[index + 4] = bytes[3];
+        self.bytecode[index + 1] = bytes[0];
```

Comment:
At some point we should check if this adds bound checking, and if we can do the bounds checking only once. Might enhance performance slightly.

Reply:
From my checking it seems that it does generate a bounds check per array access, my guess is that so it can give accurate panic locations if out-of-bounds

See godbolt assembly instructions: <https://godbolt.org/z/q8azssz5K>

Reply:
Explicitly checking beforehand and using a little unsafe we can remove the bounds checks: <https://godbolt.org/z/bKT9ozr8s>

Reply:
Sometimes in Rust, if you check the full range before all the array access with an assert, it will remove all bounds checking without needing "unsafe". We would need to test that :)

Reply:
Yeah, but through by testing it doesn't seem so do so <https://godbolt.org/z/q15WGh9n8> in this case...